compression/compression.go

```diff
@@ -0,0 +1,163 @@
+package compression
+
+import (
+	"bytes"
+	"compress/bzip2"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os/exec"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/containerd/ioutils"
+	"github.com/docker/containerd/pools"
+)
+
+type (
+	// Compression is the state represents if compressed or not.
+	Compression int
+)
+
+const (
+	// Uncompressed represents the uncompressed.
+	Uncompressed Compression = iota
+	// Bzip2 is bzip2 compression algorithm.
+	Bzip2
```

Comment:
Do we want to support compression algorithms which Docker does not use and is not included in the OCI image spec. My preference is limiting the amount of code we bring in if there is not a need.